### PR TITLE
Support React Native 0.75

### PR DESCRIPTION
**Pull request by tomekzaw**
## Summary

This PR adds support for React Native 0.75 (currently: **0.75.0-rc.6**).

https://swmansion.slack.com/archives/C048Z2NNEBY/p1719603649642539

### CODE REVIEW

**Comment by raghav-reglobe** [comment]
Hi when can we expect this to be merged.

**Comment by tomekzaw** [comment]
Hi! We still need to check this PR against 0.75.0-rc.4 and do some cleanup. I hope we can get this merged next week.

**Comment by tomekzaw** [comment]
TODO: bump example apps dependencies to versions that support RN 0.75

**Review by piaskowyk — APPROVED** [review]
`reactNativeVersionPatch` looks scary, but I understand it 💀

File: apps/common-app/src/examples/index.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ import ComposedHandlerInternalMergingExample from './ComposedHandlerInternalMerg
 import BorderRadiiExample from './SharedElementTransitions/BorderRadii';
 import FreezingShareablesExample from './ShareableFreezingExample';
 import TabNavigatorExample from './SharedElementTransitions/TabNavigatorExample';
-import StrictDOMExample from './StrictDOMExample';
+// import StrictDOMExample from './StrictDOMExample';
```
Comment (szydlovsky) [inline apps/common-app/src/examples/index.ts:133]: Do we want to keep it commented out? 😄

File: apps/common-app/src/examples/index.ts [hunk 2]
```diff
@@ -527,11 +527,11 @@ export const EXAMPLES: Record<string, Example> = {
...
-  StrictDOMExample: {
-    icon: '👮‍♂️',
-    title: 'React Strict DOM',
-    screen: StrictDOMExample,
-  },
+  // StrictDOMExample: {
+  //   icon: '👮‍♂️',
+  //   title: 'React Strict DOM',
+  //   screen: StrictDOMExample,
+  // },
```
Comment (szydlovsky) [inline apps/common-app/src/examples/index.ts:534]: Same as above

File: packages/react-native-reanimated/android/src/reactNativeVersionPatch/ReanimatedUIManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedUIManager.java [hunk 3]
```diff
@@ -186,26 +170,6 @@ public void dispatchViewManagerCommand(
...
-  @ReactMethod
-  public void showPopupMenu(int reactTag, ReadableArray items, Callback error, Callback success) {
-    super.showPopupMenu(reactTag, items, error, success);
-  }
-
-  @ReactMethod
-  public void dismissPopupMenu() {
-    super.dismissPopupMenu();
-  }
-
```
Comment (szydlovsky) [inline packages/react-native-reanimated/android/src/reactNativeVersionPatch/ReanimatedUIManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedUIManager.java:208]: Why do we delete this?

**Review by szydlovsky — COMMENTED** [review]
Need some answers first 👁️👅👁️

**Comment by tomekzaw** [comment]
TODO: investigate weird home screen layout on Fabric iOS
Screenshot 2024-07-31 at 18 06 34

File: apps/common-app/src/examples/index.ts (hunk 1 above)
Comment (tomekzaw) [reply apps/common-app/src/examples/index.ts]: Fixed in cf32307efcef94533b1a15581dc182614b51fe3a

**Review by szydlovsky — APPROVED** [review]

**Comment by raghav-reglobe** [comment]
Hi @tomekzaw, please help us publishing this, as other two libs already published.

**Comment by tomekzaw** [comment]
@raghav-reglobe We'll merge this PR and publish 3.15.0 once we fix build error in macOS example.

**Pull request merged**